### ISSUE TITLE
Abap includes not recognized as dependency

### ISSUE THREAD

**Issue by ed-holland** [abapGit/dependencies#11]
We use SAP includes in our code. These are not detected as dependency.

They are found:
 ./src/isu/#redwood#1isu.fugr.#redwood#l1isutop.abap[3, 1]                 - Include RSXPGDEF not found (check_include)
 ./src/isu/#redwood#1isu.fugr.#redwood#l1isutop.abap[4, 1]                 - Include RSADMKEY not found (check_include)
 ./src/isu/#redwood#1isu.fugr.#redwood#l1isutop.abap[5, 1]                 - Include RSXMIBAPI not found (check_include)

Can this be added?

Question: Is there a report that shows the object depency list? I have more objects from SAP missing.

**Comment by larshp** [comment]
try setting a breakpoint at https://github.com/abapGit/dependencies/blob/master/src/zcl_abapgit_deps_find.clas.abap#L73

pull requests welcome

**Comment by larshp** [comment]
try again, latest version, see changes in https://github.com/abapGit/dependencies/commit/57f106b94d94c542365571647b41f5998db52c5a

back when I first developed this program, abaplint did not check as much as it do now, so no need to add the INCLUDEs back then

**Comment by ed-holland** [comment]
Includes where found and added except `<CNTN01>` (is a valid abap name).
Also i noticed that the includes are not parsed themselves, resulting in more not founds.

I'll try to find some time tomorrow to debug.

**Comment by ed-holland** [comment]
I've done some debugging and noticed the following types which might need to be included into the dependency checks:

VIEW, TTYP, STRU

There where some others where i am not sure (depends if it can be used in visual code):

FUNC, OM, MSAG, MESS, DGT

**Comment by ed-holland** [comment]
I've added VIEW,STRU,TTYP as a test, but i think the issue is somewhere else. The linter now goes to far with the syntax checks. It should not (syntax) check the dependent objects. They are sound as they are active and part of SAP.

![image](https://user-images.githubusercontent.com/53995991/84818695-2fbeef00-b017-11ea-80d0-7da9b474c204.png)

**Comment by larshp** [comment]
if the custom code uses a SAP object that uses a SAP object, and we want full linting, then both SAP objects must be known

I think we should deprecate https://github.com/abapGit/dependencies, and move the code from it into this repository, to avoid having the same logic in multiple places. Once that is done we can add more tooling around it

**Comment by ed-holland** [comment]
Agree both should use the same dependency check as otherwise results will deviate.

**Comment by larshp** [comment]
starting this now

**Comment by larshp** [comment]
moving into https://github.com/abaplint/abaplint-abap-backend

**Comment by larshp** [comment]
@ed-holland thanks can you help closing this issue and creating in https://github.com/abaplint/abaplint-abap-backend instead

**Comment by ed-holland** [comment]
Is it merged already or still in a branch?

**Comment by larshp** [comment]
still a branch, https://github.com/abaplint/abaplint-abap-backend/pull/32 will merge in a bit and then start some refactoring

**Comment by larshp** [comment]
merged

**Comment by ed-holland** [comment]
Ran it through my projects and could not see any include related issues. It stopped after 500 messages... (still need to reduce the tests).

**Comment by ed-holland** [comment]
Found the following (using abaplint):
- issue with `Include <CNTN01> not found (check_include)`. 

Should i create issues in abaplint (one for each check)?

**Comment by larshp** [comment]
yea

**Comment by ed-holland** [comment]
this issue should be parked until the depencies are integrated into abapgit.

**Comment by larshp** [comment]
DEPRECATED
use https://github.com/abaplint/abaplint-abap-backend instead
